### ISSUE TITLE
Question: If the server is down, how to get a notification on the client?

### ISSUE THREAD

**Issue by pellea** [aspnet/SignalR#412]
Hello,

I created a client (WPF) and server (aspnetcore) sample to 'update' my old SignalR code. One requirement of my old code was to be able to know when the server is down (when the connection is closed).

I use the Microsoft.AspNetCore.SignalR.Client 1.0.0-preview1-24561 (.net 4.7).

The HubConnection.Closed event doesn't seem to work (or maybe I do not understand its purpose).
Is this the right way to achieve this?

Thanks,
Adrien.

**Comment by moozzyk** [comment]
Yes, the client should raise the Closed event when the connection is closed. If this does not work please provide more details so that we can reproduce and investigate.

**Comment by pellea** [comment]
Hello,

I tried again with the latest version (1.0.0-preview2-24979) and if I close the server before the client, I don't have the Closed event raised on the client.

```
this._connection = new HubConnection(new Uri(this._baseUrl));
this._connection.Connected += OnConnected;
this._connection.Closed += OnConnectionClosed;
```

The other Connected event is working properly.

What "more details" do you need? The code is very simple.

On the server I set up things like that:

```
app.UseSignalR(routes =>
{
        routes.MapHub<SystemManagerHub>("/mgmt");
});
```

```
public class SystemManagerHub : Hub
    { ... }
```

**Comment by moozzyk** [comment]
Please post a repro as a github repo and show how you close the server.

**Comment by pellea** [comment]
I use "CTRL-C" in VSCode.

**Comment by moozzyk** [comment]
It might be in the implementation of webSockets. I created an issue for this: https://github.com/dotnet/corefx/issues/19854

**Comment by pellea** [comment]
Please note that I have the issue in a WPF app (not UWP). When the SignalR server is closed the WPF client is still running (there is no exception thrown there).

**Comment by moozzyk** [comment]
@pellea - I was plannning to check other scenarios as well. Just hit the UWP case first. Stay tuned.

**Comment by moozzyk** [comment]
This has been fixed in cb9f44ddf6f1f5f4708016539ad66841c23c1be2
